### ISSUE TITLE
SMOG score vs SMOG index?

### ISSUE THREAD

**Issue by DonaldTsang** [shivam5992/textstat#108]
Check out https://github.com/ipeirotis/ReadabilityMetrics for some information

**Comment by alxwrd** [comment]
hi @DonaldTsang 

Thanks for raising this 🎉.

I've been looking at the implementation in the repo linked, and while there is a difference between SMOG and SMOG Index, the implementation seems to be wrong.

The Wikipedia page [SMOG_Index](https://en.wikipedia.org/wiki/SMOG_Index) redirects to [SMOG](https://en.wikipedia.org/wiki/SMOG).

--- 

Below is the **SMOG** formula, which textstat currently implements and calls it `smog_index`.

> To calculate SMOG
> 1. Count a number of sentences (at least 30)
> 2. In those sentences, count the polysyllables (words of 3 or more syllables).
 > 3. Calculate using
> ![image showing SMOG formula - grade = 1.0403 times the square root of the number of polysyllables times 30 divided by the number of sentences plus 3.1291](https://wikimedia.org/api/rest_v1/media/math/render/svg/da193073ce4254e54b077fbf56e418b14b50b9cd) 

--- 

Below is the **SMOG Index** formula, which is not currently implemented by textstat.

> [...] SMOG Index is more easily used for mental math:
> 1. Count the number of polysyllabic words in three samples of ten sentences each.
> 2. Take the square root of the nearest perfect square
> 3. Add 3

---

My thoughts here are textstat should rename `textstat.smog_index()` to `textstat.smog()`, then implement a `textstat.smog_index()` that raises a [warning](https://docs.python.org/3/library/warnings.html#warning-categories) for one version.

```python
import warnings

def smog_index(self, text):
    warnings.warn("The behaviour observed in `smog_index` is being moved to the `smog` function, please use `textstat.smog` going forward", UserWarning)
    return self.smog(text)
```

The actual SMOG Index implementation can be added after the `return` statement, where the warning and early return can be removed later.

**Comment by DonaldTsang** [comment]
Yeah that would be good, since we are making a breaking change here.

**Comment by alxwrd** [comment]
I've been researching this to find the source for the formula on the Wikipedia.

- The formula was originally published in the [Journal of reading in 1969][1], and was optimised for mental calculation.
- The author posted a revised version on [his website][2], but it has since been removed.

The original paper calls it a "Smog Grade", so my feelings are Textstat should implement `smog()` and `smog_grade()`.

[1]: McLaughlin, G. "H,(1969), SMOG grading—A new readability formula." Journal of reading 12.8: 639-649.
[2]: https://web.archive.org/web/20150905104444/http://www.harrymclaughlin.com/SMOG.htm
